### PR TITLE
First draft of Bednets

### PR DESCRIPTION
**Pull request by tbreweric**
Probably needs a bit of refining, but given the time I figured I'd put it up for comments sooner rather than later. I haven't added a final exercise as I think the material in here will be more than enough for the session.

Rather than add lots of quiz q's I've tried to focus the thought more on the outputs and linking them back to the parameters we're using and possibly trying to get them to re-run things by simply changing parameters, but that could be made more explicit in the text.

Happy for all thoughts and suggestions!

### CODE REVIEW

File: inst/tutorials/bednets/bednets.Rmd [hunk 1]
```diff
@@ -1,75 +1,211 @@
...
+library(dplyr)
+library(tidyr)
+library(ggplot2)
 knitr::opts_chunk$set(echo = FALSE)
 ```
 
+## Introduction
 
-## Topic 1
+Another widely used intervention that can be simulated using `malariasimulation` is the distribution of insecticidal bed nets (ITNs/LLINs). `malariasimulation` allows the users to tune a number of important components of both the bed nets to be distributed (e.g. efficacy, longevity) and the way in which they are distriubted (e.g. timing, coverage, and target population). In this tutorial, we will build on what we have learnt so far to learn how to simulate bed net distribution campaigns.
```
Comment (RJSheppard) [inline inst/tutorials/bednets/bednets.Rmd]:
```suggestion
Another widely used intervention that can be simulated using `malariasimulation` is the distribution of insecticide treated bed nets (ITNs/LLINs). `malariasimulation` allows the users to tune a number of important components of both the bed nets to be distributed (e.g. efficacy, longevity) and the way in which they are distributed (e.g. timing, coverage, and target population). In this tutorial, we will build on what we have learnt so far to learn how to simulate bed net distribution campaigns.
```

File: inst/tutorials/bednets/bednets.Rmd [hunk 2]
```diff
@@ -1,75 +1,211 @@
...
 ```
 
-### Exercise with Code
+Here, we have specified a bed net campaign involving two distribution events, one to occur after 1 year and another after 4 years, which will achieve a coverage of 50%. We are assuming that individuals will, on average, retain a bed net for 5 years (`retention`), and that each bed net has a half life (`gamman`) of \~960 days.
+
+```{r quiz-1}
+quiz(caption = "Parameters Qs",
+     question("What mosquito death rate (dn0) have we specified in the chunk above?",
+              answer("365"),
+              answer("0.56", correct = TRUE),
```
Comment (pwinskill) [inline inst/tutorials/bednets/bednets.Rmd]: 0.56 is rn, `dn0` is 0.533

File: inst/tutorials/bednets/bednets.Rmd [hunk 3]
```diff
@@ -1,75 +1,211 @@
...
+              answer("0.24"),
+              answer("0"), allow_retry = T),
+     question("How many columns must the matrix inputs for dn0, rn, and rnm contain if we specify bed net events to be updated 5 times (i.e. we want parameters to be updated on 5 timesteps)?",
+              answer("3"),
+              answer("Depends on the number of mosquito species"),
```
Comment (pwinskill) [inline inst/tutorials/bednets/bednets.Rmd]:
```suggestion
              answer("It depends on the number of mosquito species"),
```

File: inst/tutorials/bednets/bednets.Rmd [hunk 4]
```diff
@@ -1,75 +1,211 @@
...
+  ggplot(aes(x = timestep, y = total_mosquito_population)) +
+  geom_line(linewidth = 1.2, colour = "deepskyblue2") +
+  theme_bw() +
+  labs(x = "Time (days)", y = "Adult Female Mosquito Population Size") +
+  scale_y_continuous(limits = c(0, 7500)) +
+  geom_vline(xintercept = c(365, 1460), linetype = "dashed")
+
+# Plot the mosquito population through time:
+simulation_output |>
+  mutate(pfpr210 = n_detect_lm_730_3650 / n_730_3650) |>
```
Comment (pwinskill) [inline inst/tutorials/bednets/bednets.Rmd]: This is a `dev` malariasimulation branch naming convention. I think we should stick with the main branch, which would make it `n_detect_730_3650` I think

File: inst/tutorials/bednets/bednets.Rmd [hunk 5]
```diff
@@ -1,75 +1,211 @@
...
+# ... and view the associated helper files:
+?fun_params
 
-```{r print-limit, exercise=TRUE, exercise.eval=TRUE}
-mtcars
 ```
 
-```{r print-limit-hint}
-head(mtcars)
+Similar to `set_drugs()` in the treatment exercise, the `set_species()` function is used to load the mosquito species parameters of interest into our list of `malariasimulation` parameters. The `set_species()` function has three arguments:
```
Comment (pwinskill) [inline inst/tutorials/bednets/bednets.Rmd]: Is it possible that attendess may have jumped here before seeing `set_drugs()`
```suggestion
The `set_species()` function is used to load the mosquito species parameters of interest into our list of `malariasimulation` parameters. The `set_species()` function has three arguments:
```

**Review by pwinskill — CHANGES_REQUESTED** [review]
Really clear thanks Tom 🙏. Just a few suggestions

File: inst/tutorials/bednets/bednets.Rmd (hunk 1 above)
Comment (tbreweric) [reply inst/tutorials/bednets/bednets.Rmd]: Changes implemented as requested!

File: inst/tutorials/bednets/bednets.Rmd (hunk 2 above)
Comment (tbreweric) [reply inst/tutorials/bednets/bednets.Rmd]: Good spot - fixed!

File: inst/tutorials/bednets/bednets.Rmd (hunk 3 above)
Comment (tbreweric) [reply inst/tutorials/bednets/bednets.Rmd]: Changed!

File: inst/tutorials/bednets/bednets.Rmd (hunk 4 above)
Comment (tbreweric) [reply inst/tutorials/bednets/bednets.Rmd]: Changed!

File: inst/tutorials/bednets/bednets.Rmd (hunk 5 above)
Comment (tbreweric) [reply inst/tutorials/bednets/bednets.Rmd]: Yeah quite possible - have made the change as suggested!

**Review by pwinskill — APPROVED** [review]

**Pull request merged**